flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -879,6 +875,62 @@ else if (transitionState(current, ExecutionState.FAILED, t)) {
 		}
 	}
 
+	private void releaseNetworkResources() {
+		LOG.debug("Release task {} network resources (state: {}).", taskNameWithSubtask, getExecutionState());
+
+		for (ResultPartition partition : producedPartitions) {
+			taskEventDispatcher.unregisterPartition(partition.getPartitionId());
+		}
+
+		closeNetworkResources(producedPartitions, inputGates, isCanceledOrFailed(), getFailureCause(), taskNameWithSubtask);
+	}
+
+	private static void closeNetworkResources(
+			ResultPartition[] producedPartitions,
+			InputGate[] inputGates,
+			String taskNameWithSubtask) {
+		closeNetworkResources(producedPartitions, inputGates, false, null, taskNameWithSubtask);
+	}
+
+	/**
+	 * There are two scenarios to release the network resources. One is from {@link TaskCanceler} to early
+	 * release partitions and gates. Another is from task thread during task exiting.
+	 *
+	 * @param producedPartitions the result partitions to be closed
+	 * @param inputGates the input gates to be closed
+	 * @param isCanceledOrFailed true if the task has failed, is canceled, or is being canceled at the moment.
+	 * @param cause the exception that caused the task to fail, or null, if the task has not failed.
+	 * @param taskNameWithSubtask the name of the task, including subtask indexes
+	 */
+	private static void closeNetworkResources(
+			ResultPartition[] producedPartitions,
+			InputGate[] inputGates,
+			boolean isCanceledOrFailed,
```

Comment:
Could we rename it to just `isFailed`? Also, method `closeOrFailNetworkResources`.

Reply:
Yes, we prefer this way.